### PR TITLE
Open ended questions evaluation 35

### LINKED ISSUE #35: open ended questions evaluation

### PR DESCRIPTION
**Pull request by ayaabb**

### CODE REVIEW

File: data_access_layer/setup_mongodb.py
Comment (MohamadShaheen) [inline data_access_layer/setup_mongodb.py:1]: Please refrain from editing unnecessary files, it adds more files for me to check which takes a longer time to review a pull request which implies a delay in reviewing other pull requests. This implies for any other files whose edit is unnecessary for this issue

File: routes/basic_db_functions_route.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
-from fastapi import APIRouter
+from fastapi import APIRouter, Form
```
Comment (MohamadShaheen) [inline routes/basic_db_functions_route.py]: Consider removing unnecessary imports
```suggestion
from fastapi import APIRouter
```

File: tests/test_openai_service.py
Comment (MohamadShaheen) [inline tests/test_openai_service.py:1]: One of the tests you added adds a user stats document to the database and doesn't delete it, please consider fixing this

File: routes/openai_route.py [hunk 2]
```diff
@@ -1,18 +1,48 @@
...
+        question_text = body.question_text
+        topic = body.topic
+        difficulty = body.difficulty
+        answer = body.answer
+        evaluation_score = evaluate_answer(question=question_text, answer=answer)
+        users.add_user_stats(user_id=user_id, question_text=question_text, answer=answer, topic=topic,
+                             difficulty=difficulty,
+                             score=evaluation_score["Score"], answer_correct=(evaluation_score["Score"] >= 5),
+                             client=globals.mongo_client)
+        return evaluation_score
```
Comment (MohamadShaheen) [inline routes/openai_route.py:40]: You must return the `score and the explanation with the question and user answer` but you return the score only, please consider changing this

**Review by MohamadShaheen — CHANGES_REQUESTED** [review]

File: tests/test_openai_service.py
Comment (ayaabb) [reply tests/test_openai_service.py:1]: I fixed it in the last commit

Comment (MohamadShaheen) [reply tests/test_openai_service.py:1]: When I checked it added user stats for me in the MongoDB, please make sure to re-check it

Comment (MohamadShaheen) [reply tests/test_openai_service.py:1]: > I fixed it in the last commit

When I checked it added user stats for me in the MongoDB, please make sure to re-check it

Comment (ayaabb) [reply tests/test_openai_service.py:1]: I re-checked it one more time it is deleting user stats using 'assert check_user_existence_and_delete(data=body)'

File: routes/openai_route.py (hunk 2 above)
Comment (ayaabb) [reply routes/openai_route.py:40]: its already returning the explanation , I added question and answer

**Review by MohamadShaheen — APPROVED** [review]

**Pull request merged**